lms/djangoapps/edraak_university/tests/tests.py

```diff
@@ -0,0 +1,480 @@
+
+        user = UserFactory.create(password=password)
+        user.save()
+
+        enrollment = CourseEnrollmentFactory.create(course_id=course.id, user=user)
+        enrollment.save()
+
+        self.quick_login(user.username, password)
+
+        return self.client.get(reverse('progress', args=[unicode(course.id)]))
+
+    def test_if_tab_shown_in_response(self):
+        res = self.make_course_request(enable_university_id=True)
+        self.assertContains(res, 'University ID',
+                            msg_prefix='Advanced settings is enabled, therefore should show the tab')
+        self.assertContains(res, '/university/id/', msg_prefix='The link should appear')
+
+    def test_if_tab_is_hidden_in_response(self):
+        res = self.make_course_request(enable_university_id=False)
+        self.assertNotContains(res, 'University ID',
+                               msg_prefix='Advanced settings is disabled, therefore should NOT show the tab')
+
+        self.assertNotContains(res, '/university/id/', msg_prefix='The link should not appear')
+
+
+@ddt.ddt
+class TestHelpers(ModuleStoreTestCase):
+    """
+    Tests for the UniversityID helper functions.
+    """
+
+    def test_is_feature_enabled_helper(self):
+        with patch.dict(settings.FEATURES, EDRAAK_UNIVERSITY_APP=True):
+            self.assertTrue(helpers.is_feature_enabled(), 'Should respect the feature')
+
+        with patch.dict(settings.FEATURES, EDRAAK_UNIVERSITY_APP=False):
+            self.assertFalse(helpers.is_feature_enabled(), 'Should respect the feature')
+
+        with patch.object(settings, 'FEATURES', {}):
+            self.assertFalse(helpers.is_feature_enabled(), 'Should default to False when the feature is missing')
+
+    @ddt.unpack
+    @ddt.data(
+        {'urlconf': 'lms.urls', 'should_enable': False, 'msg': 'Should be disable when user=None in LMS'},
+        {'urlconf': 'cms.urls', 'should_enable': True, 'msg': 'Should be enabled when user=None in CMS'},
+    )
+    def test_enable_if_no_use_cms(self, urlconf, should_enable, msg):
+        with override_settings(ROOT_URLCONF=urlconf):
+            course = Mock(enable_university_id=True)
+            self.assertEquals(helpers.is_feature_enabled_on_course(course, user=None), should_enable, msg=msg)
+
+    @patch.dict(settings.FEATURES, {'EDRAAK_UNIVERSITY_APP': True, 'EDRAAK_UNIVERSITY_CSV_EXPORT': False})
+    @ddt.data(False, True)
+    def test_is_csv_export_enabled_on_course_helper_disabled(self, course_enable_university_id):
+        course = Mock(enable_university_id=course_enable_university_id)
+        is_export_enabled = helpers.is_csv_export_enabled_on_course(course)
+        self.assertFalse(is_export_enabled, 'Export should be disabled when the feature flag is')
+
+    @patch.dict(settings.FEATURES, {'EDRAAK_UNIVERSITY_APP': True, 'EDRAAK_UNIVERSITY_CSV_EXPORT': True})
+    def test_is_csv_export_enabled_on_course_helper_enabled(self):
+        self.assertFalse(
```

Comment:
what I mean, if this is done in multiple lines it would be clearer. Like 
``` 
@ddt.data((True, 'Export should be disabled when the feature flag is'), (False, 'Export should be enabled when all three features are enabled'))
@ddt.unpack
def test_name(self, enable_university_id, msg):

    is_csv_export_enabled_on_course = helpers.is_csv_export_enabled_on_course(Mock(enable_university_id=enable_university_id))
    assertEquals(is_csv_export_enabled_on_course, enable_university_id, msg=msg)
````